[Site isolation] Page and client coordinates of pointer events are always integers in cross-origin frames

lilyspiniolas opened the pull request:
#### dd60c34a7e7e2903c7094117648311e314b10496
<pre>
[Site isolation] Page and client coordinates of pointer events are always integers in cross-origin frames
<a href="https://bugs.webkit.org/show_bug.cgi?id=302550">https://bugs.webkit.org/show_bug.cgi?id=302550</a>
<a href="https://rdar.apple.com/164027526">rdar://164027526</a>

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

Update HitTestResult to store the point in the inner node frame as a DoublePoint
instead of a LayoutPoint, and update RemoteUserInputEventData to store the
transformed point as a DoublePoint instead of an IntPoint.

Tested on iOS by the following new layout tests:

iOSMouseSupport.FractionalCoordinatesInSimpleIFrameCrossOrigin
iOSMouseSupport.FractionalCoordinatesInRotatedIFrameCrossOrigin
iOSMouseSupport.FractionalCoordinatesInScaledIFrameCrossOrigin

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::prepareMouseEvent):
* Source/WebCore/dom/Document.h:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::documentPointForWindowPoint):
(WebCore::EventHandler::userInputEventDataForRemoteFrame):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::prepareMouseEvent):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::dragSourceEndedAt):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/RemoteUserInputEventData.h:
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::HitTestResult):
(WebCore::HitTestResult::operator=):
(WebCore::HitTestResult::append):
* Source/WebCore/rendering/HitTestResult.h:
(WebCore::HitTestResult::pointInInnerNodeFrame const):
(WebCore::HitTestResult::doublePointInInnerNodeFrame const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::performDragControllerAction):
(WebKit::WebPageProxy::dragEnded):
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::didPerformImmediateActionHitTest):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::potentialTapAtPosition):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::potentialTapAtPosition):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrameCrossOrigin)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrameCrossOrigin)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrameCrossOrigin)):

Canonical link: <a href="https://commits.webkit.org/303146@main">https://commits.webkit.org/303146@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e3cc13d06d2c33d5ab6a38e346a714d75cc8c78

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131441 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3773 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42405 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138947 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83209 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/94c583ae-1a17-4368-bacf-9cd1d9c91218) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133311 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3794 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3613 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/100371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134387 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2736 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117656 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81158 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/dd2a6a65-c5cf-4ac7-b097-8964e5ab4934) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2651 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/393 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82139 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111263 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35782 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141592 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3516 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36299 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/108739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3562 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3142 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108959 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27609 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2670 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113986 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56702 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3578 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32399 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3401 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66986 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3600 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3508 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
Not a full review, but the commit message mentions layout tests, I think you mean API tests??

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -161,6 +163,88 @@ void SetUp() final
+
+static std::pair<RetainPtr<TestWKWebView>, RetainPtr<TestNavigationDelegate>> siteIsolatedViewAndDelegate(const TestWebKitAPI::HTTPServer& server)
+{
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
```
```suggestion
    RetainPtr navigationDelegate = adoptNS([TestNavigationDelegate new]);
```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -161,6 +163,88 @@ void SetUp() final
+    RetainPtr configuration = server.httpsProxyConfiguration();
+    [[configuration preferences] _setSiteIsolationEnabled:YES];
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration.get()]);
```
```suggestion
    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration.get()]);
```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -354,68 +438,53 @@ void SetUp() final
 TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrame)
 {
-    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    __block RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```
`__block` is unnecessary here.
```suggestion
    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -354,68 +438,53 @@ void SetUp() final
 {
-    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
-    __block RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
+    __block RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```
```suggestion
    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -354,68 +438,53 @@ void SetUp() final
 {
-    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
-    __block RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
+    __block RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```
```suggestion
    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```

lilyspiniolas committed:
[Site isolation] Page and client coordinates of pointer events are always integers in cross-origin frames
https://bugs.webkit.org/show_bug.cgi?id=302550
rdar://164027526

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

Update HitTestResult to store the point in the inner node frame as a DoublePoint
instead of a LayoutPoint, and update RemoteUserInputEventData to store the
transformed point as a DoublePoint instead of an IntPoint.

Tested on iOS by the following new layout tests:

iOSMouseSupport.FractionalCoordinatesInSimpleIFrameCrossOrigin
iOSMouseSupport.FractionalCoordinatesInRotatedIFrameCrossOrigin
iOSMouseSupport.FractionalCoordinatesInScaledIFrameCrossOrigin

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::prepareMouseEvent):
* Source/WebCore/dom/Document.h:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::documentPointForWindowPoint):
(WebCore::EventHandler::userInputEventDataForRemoteFrame):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::prepareMouseEvent):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::dragSourceEndedAt):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/RemoteUserInputEventData.h:
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::HitTestResult):
(WebCore::HitTestResult::operator=):
(WebCore::HitTestResult::append):
* Source/WebCore/rendering/HitTestResult.h:
(WebCore::HitTestResult::pointInInnerNodeFrame const):
(WebCore::HitTestResult::doublePointInInnerNodeFrame const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::performDragControllerAction):
(WebKit::WebPageProxy::dragEnded):
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::didPerformImmediateActionHitTest):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::potentialTapAtPosition):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::potentialTapAtPosition):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrameCrossOrigin)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrameCrossOrigin)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrameCrossOrigin)):

Canonical link: https://commits.webkit.org/303146@main
